internal/envtest/environment.go

```diff
@@ -75,6 +77,52 @@ func init() {
 	utilruntime.Must(admissionv1.AddToScheme(scheme.Scheme))
 }
 
+// RunTestsInput is the input for RunTests.
+type RunTestsInput struct {
+	M                   *testing.M
+	Env                 **Environment
```

Comment:
I didn't find a way to get this to work with just one pointer. The problem is that I have to change 
```go
var (
	env *envtest.Environment
)
```
in the `suite_test.go` files to
```go
var (
	env envtest.Environment
)
```
and then there are all kinds of linter findings because envtest.Environment contains a mutex which cannot be used with funcs with value receivers

Reply:
Thanks for the explanation. 
Let's just add a comment explaining that the double pointer is required due to the mutex embedded in Environment

Reply:
Done

Reply:
Okay whatever I did before. I think I got it to work. Not sure what the difference is to before

Reply:
Okay it really doesn't work. I found a way to have a non-nil env at the call site but every field of the env is nil (e.g. the client)

Reply:
I'l resolve it as it's solved by Vince's changes